core/ir/encode_api.h

```diff
@@ -51,6 +51,38 @@ typedef enum _dr_isa_mode_t {
     DR_ISA_ARM_THUMB,         /**< Thumb (ARM T32). */
     DR_ISA_ARM_A64,           /**< ARM A64 (AArch64). */
     DR_ISA_RV64IMAFDC,        /**< RISC-V (rv64imafdc). */
+    DR_ISA_REGDEPS,
+    /**<
+     * This is a synthetic ISA that has the purpose of preserving register dependencies
+     * and giving hints on the type of operations each instruction is performing.
+     *
+     * For this reason the majority of operations that would normally work on instructions
+     * coming from an actual ISA (e.g., #DR_ISA_AMD64) are not supported.
+     *
+     * Currently we support:
+     * - instr_convert_to_isa_regdeps(), which converts an #instr_t of an actual ISA to a
+     *   #DR_ISA_REGDEPS instruction;
+     * - instr_encode() and instr_encode_to_copy(), to encode a #DR_ISA_REGDEPS #instr_t
+     *   into a sequence of contiguous bytes;
+     * - decode() and decode_from_copy(), to decode an encoded #DR_ISA_REGDEPS instruction
+     *   into an #instr_t.
+     *
+     * A #DR_ISA_REGDEPS #instr_t holds only the encoded information described in the
+     * encoding scheme, which is:
+     * - categories (from #dr_instr_category_t), to indicate the type of operation
+     *   performed (e.g., load, store, floating point math operation, branch, etc.);
+     * - arithmetic flags, with no distinction between different flags, we only report if
+     *   at least one flag was read and/or written;
+     * - number of source and destination register operands;
+     * - operation size;
+     * - list of register operand identifiers, separated in source and destination;
+     * - ISA mode, which is #DR_ISA_REGDEPS.
+     *
+     * Querying additional information (e.g., the instruction opcode) will return the
+     * default value generated by how the instruction was created.  If instr_create() was
+     * used, then zero will be returned.  If the #instr_t was allocated on the stack, then
+     * all values not described above are undefined.
```

Comment:
These last two sentences can probably be removed and replaced with one saying it's zero: the API requires calling instr_init() for stack vars -- there won't be uninit values floating around.

Reply:
Done.